### PR TITLE
Support monitoring via opencensus

### LINKED ISSUE #11: mongen: Use go.opencensus.io/stats for metrics
See https://godoc.org/go.opencensus.io/stats for API docs.

Generated code could look like:

```go
package servicemws

import (
	"context"
	"time"

	"go.opencensus.io/stats"
)

type monitoringService struct {
	next        Service
	totalOps    *stats.Int64Measure
	failedOps   *stats.Int64Measure
	opsDuration *stats.Float64Measure
}

func (m *monitoringService) Work(ctx context.Context) error {
	stats.Record(ctx, m.totalOps.M(1))
	start := time.Now()
	err := m.next.Work(ctx)
	stats.Record(ctx, m.opsDuration.M(time.Since(start).Seconds()))
	if err != nil {
		stats.Record(ctx, m.failedOps.M(1))
	}
	return err
}
```

### PR DESCRIPTION
**Pull request by Makpoc**
Fixes #11

### CODE REVIEW

**Comment by Bo0mer** [comment]
Hello and thanks for the PR!

I'll try it out and will post any comments that I have regarding the change.

Thanks,
Ivan

File: pkg/astgen/method.go [hunk 1]
```diff
@@ -26,6 +26,11 @@ func (m *Method) SetType(funcType *ast.FuncType) {
 func (m *Method) AddStatement(stmt ast.Stmt) {
 	m.statements = append(m.statements, stmt)
 }
+func (m *Method) AddStatements(stmts []ast.Stmt) {
+	for _, stmt := range stmts {
```
Comment (Bo0mer) [inline pkg/astgen/method.go]: You could rather do:
```
	m.statements = append(m.statements, stmts...)
```

File: cmd/mongen/internal/opencensus/oc_model.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	timePkg    string
+	contextPkg string
+	tagPkg     string
+}
+
+type opencensusModel struct {
+	fileBuilder *astgen.File
+	structName  string
+
+	packageAliases packageAliases
```
Comment (Bo0mer) [inline cmd/mongen/internal/opencensus/oc_model.go:22]: WDYT about embedding this, so it's shorter to use the members? I.e.:

```go
// packageAliases holds the aliases of all imported packages in the generated source file.
type packageAliases struct {
	timePkg    string
	contextPkg string
	tagPkg     string
	statsPkg   string
	sourcePkg  string
}

type opencensusModel struct {
	fileBuilder *astgen.File
	structName  string

	packageAliases
}

func newOpencensusModel(interfacePath, interfaceName, structName, targetPkg string) *opencensusModel {
	file := astgen.NewFile(targetPkg)
	m := &opencensusModel{
		fileBuilder: file,
		structName:  structName,
		packageAliases: packageAliases{
			timePkg:    file.AddImport("", "time"),
			contextPkg: file.AddImport("", "context"),
			tagPkg:     file.AddImport("", "go.opencensus.io/tag"),
			statsPkg:   file.AddImport("", "go.opencesus.io/stats"),
			sourcePkg:  file.AddImport("", interfacePath),
		},
	}

	strct := astgen.NewStruct(structName)
	strct.AddField("next", m.sourcePkg, interfaceName)
	strct.AddFieldWithType(totalOps, pointerExpr(m.statsPkg, "Int64Measure"))
	strct.AddFieldWithType(failedOps, pointerExpr(m.statsPkg, "Int64Measure"))
	strct.AddFieldWithType(opsDuration, pointerExpr(m.statsPkg, "Float64Measure"))
	strct.AddFieldWithType(ctxFuncName, buildCtxFuncType(m.contextPkg))
	file.AppendDeclaration(strct)

	constructorBuilder := newOCConstructorBuilder(
		m.statsPkg, m.contextPkg, m.sourcePkg, interfaceName)
	file.AppendDeclaration(constructorBuilder)

	return m
}
```

Other changes that I've introduced are:
* packages are sorted so std lib ones are first
* the model struct is initialised in a single construct

File: cmd/mongen/internal/common/builders.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
+package common
```
Comment (Bo0mer) [inline cmd/mongen/internal/common/builders.go]: I'm not a huge fan of `common` packages, as they tend to become bloated with everything. Do you find it reasonable to keep this code in the `cmd/mongen`'s `main` package?

File: cmd/mongen/internal/opencensus/oc_model.go (hunk 2 above)
Comment (Makpoc) [reply cmd/mongen/internal/opencensus/oc_model.go:22]: Embedding does make it shorter, but I like the explicit grouping under packageAliases when accessing the *Pkg properties.

:+1: for sorting the imports and changes to init.

Let me know how you feel about leaving the packageAliases as separate struct member tho.

File: cmd/mongen/internal/common/builders.go (hunk 3 above)
Comment (Makpoc) [reply cmd/mongen/internal/common/builders.go]: I understand. The `common` here was intended to mark the builders that are common for both `gokit` and `opencensus` and not as a _common dumping ground_. I'll see how i can move things around to remove the need for this package.

File: cmd/mongen/internal/opencensus/oc_model.go (hunk 2 above)
Comment (Bo0mer) [reply cmd/mongen/internal/opencensus/oc_model.go:22]: Sure, I'm fine. Explicit is better than implicit.

File: cmd/mongen/internal/common/builders.go (hunk 3 above)
Comment (Makpoc) [reply cmd/mongen/internal/common/builders.go]: `common` is not split into `commonbuilders` and `pkg/transformation`

File: cmd/mongen/internal/opencensus/oc_model.go (hunk 2 above)
Comment (Makpoc) [reply cmd/mongen/internal/opencensus/oc_model.go:22]: Cool. I've applied your other comments
* sort imports
* init the model in a single construct

**Pull request merged**